### ISSUE TITLE
Emit warning for standalone pre/post increment/decrement operations with 'set -e'

### ISSUE THREAD

**Issue by varungandhi-src** [koalaman/shellcheck#2777]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

set -e

f() {
        local a=0
        ((a++))
}

f
```

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

```
(warning): ((a++)) may fail if a is 0, which may be counter-intuitive 
```

There could be a hint to optionally add a `|| :`. There should be a similar warning for a standalone pre-increment, as well as corresponding decrement operations. I'm not picky about the exact wording, but I think there should be a warning here.

I think this is worth warning about, as in most other PLs, arithmetic operations like increment and decrement don't ever fail.

Related StackOverflow questions with vote counts at the time of writing, offered as evidence of this behavior being counter-intuitive:
- [Bash exit status of shorthand increment notation](https://stackoverflow.com/questions/22428894/bash-exit-status-of-shorthand-increment-notation) (+ 15)
- [Incrementing a variable triggers EXIT in bash 4, but not in bash 3](https://stackoverflow.com/questions/6877012/incrementing-a-variable-triggers-exit-in-bash-4-but-not-in-bash-3) (+ 15)
- [Why does ((count++)) return 1 exit code first time run](https://stackoverflow.com/questions/41349109/why-does-count-return-1-exit-code-first-time-run) (+ 8)
- [Why does bash post-increment operator set a nonzero result code? [duplicate]](https://stackoverflow.com/questions/67798310/why-does-bash-post-increment-operator-set-a-nonzero-result-code)
- [Exit status 1 set after postfix increment from 0 to 1 [duplicate]](https://stackoverflow.com/questions/65334750/exit-status-1-set-after-postfix-increment-from-0-to-1)

**Comment by ambroisie** [comment]
I got bitten by this today, had to run with `set -x` to understand what the issue was.

I think either suggesting `|| :` or `: $((var++))` would be fine in this case.

**Comment by auspicacious** [comment]
I ran into this recently when someone gave me a script that ran fine on their MacOS machine, but wouldn't work on my Linux machine. As far as I can tell, the ancient version of Bash that ships with MacOS doesn't have the "correct" behavior here and doesn't fail. (Worse, AI agents are apparently writing code that includes `((var++))`, since I know that one wrote much of this script).

It might be better to suggest using `((var+=1))` instead, since this doesn't distract someone reading the code with syntax that isn't related to the operation being performed.
